File: src/main/java/components/Sprite.java
package components;

import org.joml.Vector2f;
import renderer.Texture;
import util.AssetPool;

public class Sprite {
    private Texture texture;
    private Vector2f[] texCoords;

    public Sprite(Texture texture) {
        init(texture);
    }

    private void init(Texture texture) {
        this.texture = texture;
        this.texCoords = new Vector2f[]{
                new Vector2f(1, 1),
                new Vector2f(1, 0),
                new Vector2f(0, 0),
                new Vector2f(0, 1),
        };

    }

    public Sprite(String assetPoolKey) {
        init(AssetPool.getTexture(assetPoolKey));
    }

    public Sprite(Texture texture, Vector2f[] texCoords) {
        this.texture = texture;
        this.texCoords = texCoords;
    }

    public Texture getTexture() {
        return texture;
    }

    public Vector2f[] getTexCoords() {
        return texCoords;
    }
}

File: src/main/java/components/Sprite.kt
package components

import org.joml.Vector2f
import renderer.Texture

class Sprite {
    private var width = 0f
    private var height = 0f
    var texture: Texture? = null
        private set
    var texCoords: Array<Vector2f> = arrayOf(
        Vector2f(1f, 1f),
        Vector2f(1f, 0f),
        Vector2f(0f, 0f),
        Vector2f(0f, 1f)
    )
        private set

    fun setTexCoords(texCoords: Array<Vector2f>) {
        this.texCoords = texCoords
    }

    fun setTexture(texture: Texture) {
        this.texture = texture
    }

    fun getWidth(): Float {
        return width
    }

    fun getHeight(): Float {
        return height
    }

    fun getTexID(): Int {
        return texture?.getID() ?: -1
    }

    fun setWidth(width: Float) {
        this.width = width
    }

    fun setHeight(height: Float) {
        this.height = height
    }

}